nexus/src/app/mod.rs

```diff
@@ -647,28 +653,33 @@ impl Nexus {
         self.producer_server.lock().unwrap().replace(producer_server);
     }
 
+    /// Terminates all servers.
+    ///
+    /// This function also waits for the servers to shut down.
     pub(crate) async fn close_servers(&self) -> Result<(), String> {
         // NOTE: All these take the lock and swap out of the option immediately,
         // because they are synchronous mutexes, which cannot be held across the
         // await point these `close()` methods expose.
         let external_server = self.external_server.lock().unwrap().take();
+        let mut res = Ok(());
+
         if let Some(server) = external_server {
-            server.close().await?;
+            res = res.and(server.close().await);
```

Comment:
Technically, this isn't necessary as a part of this PR -- but i was debating putting the datastore termination here.

I ended up opting against it, but still think it's useful to "try to shut down all servers" even if one of them fails to terminate.